pkgs/pkgs-lib/formats/libconfig/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, callPackage
+, runCommandCC
+, rustPlatform
+, libconfig
+, ...
+}:
+let
+  # Implementation notes:
+  #   Libconfig spec: https://hyperrealm.github.io/libconfig/libconfig_manual.html
+  #
+  #   Since libconfig does not allow setting names to start with an underscore,
+  #   this is used as a prefix for both special types and include directives.
+  #
+  #   The difference between 32bit and 64bit values became optional in libconfig
+  #   1.5, so we assume 64bit values for all numbers.
+
+  libconfig-generator = rustPlatform.buildRustPackage {
+    name = "libconfig-generator";
+    version = "0.1.0";
+    src = ./.;
+
+    passthru.updateScript = ./update.sh;
+
+    cargoLock.lockFile = ./Cargo.lock;
+  };
+
+  libconfig-validator = runCommandCC "libconfig-validator"
+    {
+      buildInputs = [ libconfig ];
+    }
+    ''
+      $CC -lconfig -x c - -o "$out" ${./validator.c}
+    '';
```

Comment:
These two derivations won't work with cross compilation.

I see two ways to fix it. Either
 - Turns these into proper packages, perhaps with `-nixos` suffix to indicate that they're unofficial but rather serve the (wider) NixOS project.
   - Also you could talk to upstream about making commands like these official. You'd help more people and consistency with the "outside world" is nice. Optional.
 - Or take everything from `buildPackages` (`buildPackages.rustPlatform`, `buildPackages.runCommandCC`, etc)

Reply:
> Turns these into proper packages.

This would certainly make it easier to run the update script, but the packages themselves are a bit weird to use standalone. Maybe if we clean up the cli interface a bit?

> making commands like these official.

I'm a bit unsure about what you mean here. libconfig doesn't seem to have any existing cli interface, it's just a library and nothing more. Are you proposing we ask whether they'd be interested in making a cli interface to host the `validate` command?

I think I'll use `buildPackages` for now. Might move it later.